crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -207,20 +207,13 @@ impl NotificationClient {
 
                         tokio::time::sleep(Duration::from_millis(wait)).await;
 
-                        let new_event = room.decrypt_event(raw_event.cast_ref()).await?;
-
-                        if !is_event_encrypted(
-                            new_event
-                                .event
-                                .deserialize()
-                                .map_err(|_| Error::InvalidRumaEvent)?
-                                .event_type(),
-                        ) {
-                            trace!("Waiting succeeded!");
-                            return Ok(Some(new_event));
-                        }
+                        let Ok(new_event) = room.decrypt_event(raw_event.cast_ref()).await else {
```

Comment:
Can we check if it's a missing key error and still bubble up the error if not?

Reply:
Good idea. At first I thought the only cause of failure could be decryption errors, but I was very far from the truth: deserialization can fail, reads to the stores, etc. New code should be more targeted.